src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpMetricSerializer.cs

```diff
@@ -0,0 +1,352 @@
+        }
+
+        writePosition = WriteResourceMetrics(buffer, writePosition, resource, ScopeMetricsList);
+        ReturnMetricListToPool();
+
+        return writePosition;
+    }
+
+    private static void ReturnMetricListToPool()
+    {
+        if (ScopeMetricsList.Count != 0)
+        {
+            foreach (var entry in ScopeMetricsList)
+            {
+                entry.Value.Clear();
+                MetricListPool.Push(entry.Value);
+            }
+
+            ScopeMetricsList.Clear();
+        }
+    }
+
+    private static int WriteResourceMetrics(byte[] buffer, int writePosition, in Resources.Resource? resource, in Dictionary<string, List<Metric>> scopeMetrics)
+    {
+        writePosition = ProtobufOtlpResourceSerializer.WriteResource(buffer, writePosition, resource);
+        writePosition = WriteScopeMetrics(buffer, writePosition, scopeMetrics);
+
+        return writePosition;
+    }
+
+    private static int WriteScopeMetrics(byte[] buffer, int writePosition, in Dictionary<string, List<Metric>> scopeMetrics)
+    {
+        if (scopeMetrics != null)
+        {
+            foreach (KeyValuePair<string, List<Metric>> entry in scopeMetrics)
+            {
+                writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ResourceMetrics_Scope_Metrics, ProtobufWireType.LEN);
+                int resourceMetricsScopeMetricsLengthPosition = writePosition;
+                writePosition += ReserveSizeForLength;
+
+                writePosition = WriteScopeMetric(buffer, writePosition, entry.Value);
+
+                ProtobufSerializer.WriteReservedLength(buffer, resourceMetricsScopeMetricsLengthPosition, writePosition - (resourceMetricsScopeMetricsLengthPosition + ReserveSizeForLength));
+            }
+        }
+
+        return writePosition;
+    }
+
+    private static int WriteScopeMetric(byte[] buffer, int writePosition, in List<Metric> metrics)
+    {
+        writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ScopeMetrics_Scope, ProtobufWireType.LEN);
+        int instrumentationScopeLengthPosition = writePosition;
+        writePosition += ReserveSizeForLength;
+
+        Debug.Assert(metrics.Any(), "Metrics collection is not expected to be empty.");
+        var meterName = metrics[0].MeterName;
+        var meterVersion = metrics[0].MeterVersion;
+        var meterTags = metrics[0].MeterTags;
+
+        writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.InstrumentationScope_Name, meterName);
+        if (meterVersion != null)
+        {
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.InstrumentationScope_Version, meterVersion);
+        }
+
+        if (meterTags != null)
+        {
+            // TODO: DON'T HAVE ANY TESTS THAT EVALUATE THIS

```

Comment:
Does this mean current solution does not support instrumentation scope?

Reply:
No, this simply means that we don't have any unit tests validating the Instrumentation Scope Attributes are serialized correctly.
I'm planning on expanding the test coverage in a follow up PR.